executor/builder.go

```diff
@@ -766,6 +768,21 @@ func (b *executorBuilder) buildLoadStats(v *plannercore.LoadStats) Executor {
 	return e
 }
 
+func (b *executorBuilder) buildIndexAdvise(v *plannercore.IndexAdvise) Executor {
+	e := &IndexAdviseExec{
+		baseExecutor: newBaseExecutor(b.ctx, nil, v.ExplainID()),
+		IsLocal:      v.IsLocal,
```

Comment:
Since this field is a pre-requisite, can we do it in parser and remove this field from planner / executor?

Reply:
Any suggestions for this one? @eurekaka

Reply:
How about replacing `LocalOpt` in https://github.com/pingcap/parser/blob/12ed74961445d1019ea2f9ce85febb75a87c3c59/parser.y#L10969 to literal "LOCAL"? I guess the reason that `load data` checks `IsLocal` in tidb layer instead of parser layer is for compatibility of `load data infile` syntax, but for `advise index` we have no such concern. I am fine with current code actually, but I think it would be better to remove it. @lamxTyler Any comments?

Reply:
The difference is only where the error will be throw, in parser or executor, since it does not matter too much and changing the parser requires a new pr, I prefer not change it...

Reply:
> The difference is only where the error will be throw, in parser or executor, since it does not matter too much and changing the parser requires a new pr, I prefer not change it...

Agree...